pageserver/src/tenant/block_io.rs

```diff
@@ -76,6 +68,30 @@ impl<'a> Deref for BlockLease<'a> {
     }
 }
 
+pub(crate) enum BlockReaderRef<'a> {
```

Comment:
Could add a quick note on what is the purpose of this. If we read this many months from now it might look a bit extra.

```suggestion
/// Provides the ability to read blocks from different sources, alternative
/// to using traits which are problematic for asyncification.
pub(crate) enum BlockReaderRef<'a> {
```

Or something like this.